horreum-java-client/src/test/java/io/hyperfoil/tools/HorreumTestBase.java

```diff
@@ -0,0 +1,161 @@
+    protected static boolean CREATE_HORREUM_TEST;
+    protected static boolean HORREUM_DUMP_LOGS;
+
+    protected static HorreumClient horreumClient;
+
+    protected static Test newTest;
+
+    private static Logger log = Logger.getLogger(HorreumTestBase.class);
+
+    static {
+        configProperties = new Properties();
+        InputStream propertyStream = Thread.currentThread().getContextClassLoader().getResourceAsStream("env.properties"); //TODO: make configurable
+        try {
+            if (propertyStream != null) {
+                configProperties.load(propertyStream);
+
+                HORREUM_KEYCLOAK_BASE_URL = getProperty("horreum.keycloak.base-url");
+                HORREUM_BASE_URL = getProperty("horreum.base-url");
+                HORREUM_CLIENT_ID = getProperty("horreum.client-id");
+                HORREUM_KEYCLOAK_REALM = getProperty("keycloak.realm");
+
+                HORREUM_USERNAME = getProperty("horreum.username");
+                HORREUM_PASSWORD = getProperty("horreum.password");
+                HORREUM_CLIENT_SECRET = getProperty("horreum.client_secret");
+
+                START_HORREUM_INFRA = Boolean.valueOf(getProperty("horreum.start-infra"));
+                HORREUM_DUMP_LOGS = Boolean.valueOf(getProperty("horreum.dump-logs"));
+                CREATE_HORREUM_TEST = Boolean.valueOf(getProperty("horreum.create-test"));
+            } else {
+                throw new RuntimeException("Could not load test configuration");
+            }
+        } catch (IOException ioException) {
+            throw new RuntimeException("Failed to load configuration properties");
+        }
+    }
+
+    public static DockerComposeContainer environment = null;
+
+    protected io.hyperfoil.tools.horreum.entity.json.Test getExistingtest(){
+         return horreumClient.testService.get(newTest.id, null);
+    }
+
+    protected static String getProperty(String propertyName) {
+        return configProperties.getProperty(propertyName).trim();
+    }
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        initialiseRestClients();
+
+        log.info("Starting Infra: " + START_HORREUM_INFRA);
+        if (START_HORREUM_INFRA) {
+
+            environment = new DockerComposeContainer(new File("src/test/resources/testcontainers/docker-compose.yml"))
+                    .withLocalCompose(true)
+                    .withExposedService("postgres_1", 5432)
+            ;
+            environment.start();
+            //wait for Horreum infrastructure to spin up
+            log.info("Waiting for Horreum infrastructure to start");
+            Optional<ContainerState> optionalContainer = environment.getContainerByServiceName("horreum_1");
+            if (optionalContainer.isPresent()) {
+                ContainerState horreumState = optionalContainer.get();
+                while (!horreumState.getLogs(OutputFrame.OutputType.STDOUT).contains("started in")) {
```

Comment:
You could make this more responsive with
```
.withLogConsumer(new ToStringConsumer() {
   void accept(OutputFrame frame) {
      super.accept(frame);
      if (toUtf8String().contains("started in")) {
         latch.countDown();
      }
   }
})
```

Reply:
yes, this is suboptimal :)